File: Day70/Part3/item_durability.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int simulate(const vector<int>& a) {
    vector<int> stack;
    for (int x : a) {
        if (!stack.empty() && stack.back() == x) {
            stack.pop_back();
        } else {
            stack.push_back(x);
        }
    }
    return stack.size();
}

int main() {
    int n;
    if (cin >> n) {
        vector<int> a(n);
        for (int i = 0; i < n; ++i) cin >> a[i];

        if (n == 0) {
            cout << 0 << endl;
            return 0;
        }

        int min_remaining = n; // Upper bound

        // Try removing each item
        for (int i = 0; i < n; ++i) {
            vector<int> temp;
            temp.reserve(n - 1);
            for (int j = 0; j < n; ++j) {
                if (i == j) continue;
                temp.push_back(a[j]);
            }
            int remaining = simulate(temp);
            if (remaining < min_remaining) {
                min_remaining = remaining;
            }
        }

        cout << min_remaining << endl;
    }
    return 0;
}

File: Day70/Part3/item_durability.py
def simulate(a):
    stack = []
    for x in a:
        if stack and stack[-1] == x:
            stack.pop()
        else:
            stack.append(x)
    return len(stack)

def solve():
    try:
        line = input().split()
        if not line: return
        n = int(line[0])
        
        a = list(map(int, input().split()))
        
        if n == 0:
            print(0)
            return
            
        min_remaining = n
        
        for i in range(n):
            temp = a[:i] + a[i+1:]
            remaining = simulate(temp)
            if remaining < min_remaining:
                min_remaining = remaining
                
        print(min_remaining)
    except EOFError:
        pass

if __name__ == "__main__":
    solve()
